questionable/indexing.nim

```diff
@@ -1,16 +1,23 @@
 import std/macros
 
-macro `.?`*(expression: seq | string, brackets: untyped{nkBracket}): untyped =
+macro `.?`*(expression: seq | string | openArray, brackets: untyped{nkBracket}): untyped =
   # chain is of shape: (seq or string).?[index]
   let index = brackets[0]
   quote do:
     block:
       type T = typeof(`expression`[`index`])
-      let evaluated = `expression`
-      if `index` < evaluated.len:
-        evaluated[`index`].some
+
+      when typeof(`expression`) is openArray:
```

Comment:
Why do you want to handle openArray different here?

Reply:
I explained the issue in a [comment](https://github.com/codex-storage/questionable/pull/52#discussion_r1399164751) below.